tests/BuildSystem/Build/working-directory.llbuild

```diff
@@ -0,0 +1,38 @@
+# Check that, by default, directories are automatically created for output files.
+#
+# RUN: rm -rf %t.build
+# RUN: mkdir -p %t.build
+# RUN: cp %s %t.build/build.llbuild
+# RUN: %{llbuild} buildsystem build --serial --chdir %t.build > %t.out
+# RUN: test -d %t.build/subdir
+# RUN: test -f %t.build/subdir/subsubdir/file
+# RUN: test -L %t.build/subdir-2/subsubdir/symlink
+# RUN: %{FileCheck} --input-file=%t.out %s
+#
+# CHECK: TOUCH
+# CHECK: SYMLINK
+#
+# REQUIRES: platform=Darwin
+
+client:
+  name: basic
+
+targets:
+  "": ["<all>"]
+
+commands:
+  C.all:
+    tool: phony
+    inputs: ["subdir/subsubdir/file", "subdir-2/subsubdir/symlink"]
+    outputs: ["<all>"]
+  C.touch:
+    tool: shell
+    description: TOUCH
+    args: ["touch", "file"]
+    working-directory: "subdir/subsubdir"
+    outputs: ["subdir/subsubdir/file"]
+  C.symlink:
+    tool: symlink
+    description: SYMLINK
+    outputs: ["subdir-2/subsubdir/symlink"]
+    contents: ignored
```

Comment:
This test feels like it could safely be simplified?

Reply:
I’m sure.  I (clearly) just copied and tweaked an existing test.  I do like that it covers changing between working directories, but I didn’t evaluate it finely.